src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatabilityAnalyzer.cs

```diff
@@ -0,0 +1,446 @@
+                                    AttributeVersionsMatch(attribute.AttributeType, attribute.Version, info.Version))
+                                {
+                                    return true;
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+            return false;
+        }
+
+        private static void ReportDiagnosticsForAll(PooledDictionary<IOperation, ImmutableArray<PlatformAttributeInfo>> platformSpecificOperations, OperationBlockAnalysisContext context)
+        {
+            foreach (var platformSpecificOperation in platformSpecificOperations)
+            {
+                ReportDiagnostics(platformSpecificOperation.Key, platformSpecificOperation.Value.FirstOrDefault(), context);
+            }
+        }
+
+        private static void ReportDiagnostics(IOperation operation, PlatformAttributeInfo attribute, OperationBlockAnalysisContext context)
+        {
+            string operationName = string.Empty;
+
+            if (operation is IInvocationOperation iOperation)
+            {
+                operationName = iOperation.TargetMethod.Name;
+            }
+            else if (operation is IObjectCreationOperation cOperation)
+            {
+                operationName = cOperation.Constructor.Name;
+            }
+            else if (operation is IPropertyReferenceOperation pOperation)
+            {
+                operationName = pOperation.Property.Name;
+            }
+            else if (operation is IFieldReferenceOperation fOperation)
+            {
+                operationName = fOperation.Field.Name;
+            }
+            if (operation is IDelegateCreationOperation dOperation)
+            {
+                if (dOperation.Target is IMethodReferenceOperation mrOperation)
+                    operationName = mrOperation.Method.Name;
+            }
+            else if (operation is IEventReferenceOperation eOperation)
+            {
+                operationName = eOperation.Event.Name;
+            }
+
+            context.ReportDiagnostic(operation.CreateDiagnostic(SelectRule(attribute.AttributeType),
+                operationName, attribute.PlatformName, attribute.Version.ToString()));
+        }
+
+        private static void AnalyzeInvocationOperation(IOperation operation, OperationAnalysisContext context,
+            ref PooledDictionary<IOperation, ImmutableArray<PlatformAttributeInfo>> platformSpecificOperations)
+        {
+            using var builder = ArrayBuilder<PlatformAttributeInfo>.GetInstance();
+            AttributeData[]? attributes = null;
+
+            if (operation is IInvocationOperation iOperation)
+            {
+                attributes = FindAllPlatformAttributesApplied(iOperation.TargetMethod.GetAttributes(), iOperation.TargetMethod.ContainingType);
```

Comment:
Can `FindAllPlatformAttributesApplied` take `ISymbol` instead of attributes?
```suggestion
                attributes = FindAllPlatformAttributesApplied(iOperation.TargetMethod, iOperation.TargetMethod.ContainingType);
```

Reply:
Actually a better approach might be to change this to a switch expression that computes the underlying `ISymbol` and then invoke `var attributes = FindAllPlatformAttributesApplied(symbol);`. `FindAllPlatformAttributesApplied` should be changed to take an `ISymbol` parameter that computes `symbol.GetAttributes()` and `symbol.ContainingType`